### PR TITLE
fix: wasm addFunction tests

### PR DESCRIPTION
**Pull request by ianpurvis**
This fixes breakage in emsdk 2.0 and starts building and testing addFunction during CI.

### CODE REVIEW

**Comment by kripken** [comment]
What does `ADD_FUNCTION_SUPPORT` do? If it just enables Table growth, we can probably enable that all the time, it has no significant downside.

**Comment by ianpurvis** [comment]
`ADD_FUNCTION_SUPPORT` adds these link flags for emcc:
```
    -s RESERVED_FUNCTION_POINTERS=20
    -s EXTRA_EXPORTED_RUNTIME_METHODS=["addFunction"])
```

I like the idea of enabling this by default, it seems popular... I can go ahead and tweak `CMakeLists.txt`

**Comment by ianpurvis** [comment]
All set... 👍👍

**Comment by ianpurvis** [comment]
Sorry, things got messy when I merged #331... this should be good to go now.

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,10 +41,12 @@ set(EMCC_ARGS
   -O3
   -s ALLOW_MEMORY_GROWTH=${ALLOW_MEMORY_GROWTH}
   -s EXPORTED_RUNTIME_METHODS=["UTF8ToString"]
+  -s EXTRA_EXPORTED_RUNTIME_METHODS=["addFunction"]
   -s EXPORT_NAME="Ammo"
   -s MODULARIZE=1
   -s NO_EXIT_RUNTIME=1
   -s NO_FILESYSTEM=1
+  -s RESERVED_FUNCTION_POINTERS=20
```
Comment (kripken) [inline CMakeLists.txt:49]: this should not be needed anymore. instead use `-s ALLOW_TABLE_GROWTH`

File: CMakeLists.txt [hunk 2]
```diff
@@ -41,10 +41,12 @@ set(EMCC_ARGS
   -O3
   -s ALLOW_MEMORY_GROWTH=${ALLOW_MEMORY_GROWTH}
   -s EXPORTED_RUNTIME_METHODS=["UTF8ToString"]
+  -s EXTRA_EXPORTED_RUNTIME_METHODS=["addFunction"]
```
Comment (kripken) [inline CMakeLists.txt:44]: is this needed? where is addFunction called from?

Comment (ianpurvis) [reply CMakeLists.txt:44]: If I understand correctly, this is needed to create a function pointer... Here's an example with `btDyamicsWorld.setInternalTickCallback`

```js
function callback(world, timestep) {}

const pointer = Ammo.addFunction(callback, Ammo.INTERNAL_TICK_CALLBACK_SIGNATURE)

world.setInternalTickCallback(pointer)
```

File: CMakeLists.txt (hunk 1 above)
Comment (ianpurvis) [reply CMakeLists.txt:49]: Sounds good- should we do that for wasm and javascript builds?

File: CMakeLists.txt (hunk 2 above)
Comment (kripken) [reply CMakeLists.txt:45]: Ah yes, makes sense.

**Pull request merged**